### PR TITLE
Add translation to integration setup failures

### PR DESCRIPTION
**Pull request by gjohansson-ST**
## Proposed change

Add possibility for translations for Config Entry setup errors
Core PR: https://github.com/home-assistant/core/pull/106305

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/config/integrations/ha-config-integration-page.ts [hunk 1]
```diff
@@ -550,9 +550,20 @@ class HaConfigIntegrationPage extends SubscribeMixin(LitElement) {
...
       if (item.reason) {
         this.hass.loadBackendTranslation("config", item.domain);
-        stateTextExtra = html`${this.hass.localize(
+        const stateTextError = this.hass.localize(
           `component.${item.domain}.config.error.${item.reason}`
-        ) || item.reason}`;
+        );
+        let stateTextReason: string | null = null;
+        if (item.reason_translation_key) {
+          this.hass.loadBackendTranslation("exceptions", item.domain);
```
Comment (bramkragten) [inline src/panels/config/integrations/ha-config-integration-page.ts:558]: you will need to await this

Comment (gjohansson-ST) [reply src/panels/config/integrations/ha-config-integration-page.ts:558]:
```
'await' expressions are only allowed within async functions and at the top levels of modules.ts(1308)
ha-config-integration-page.ts(532, 11): Did you mean to mark this function as 'async'?
```
Not sure what that means and also we have loading translations on line 552 which is not awaited so not sure I would understand the difference

Comment (bramkragten) [reply src/panels/config/integrations/ha-config-integration-page.ts:558]: Yeah that seems wrong too, one way to fix it is:

```diff
@@ -549,10 +550,15 @@ class HaConfigIntegrationPage extends SubscribeMixin(LitElement) {
         `ui.panel.config.integrations.config_entry.state.${item.state}`,
       ];
       if (item.reason) {
-        this.hass.loadBackendTranslation("config", item.domain);
-        stateTextExtra = html`${this.hass.localize(
-          `component.${item.domain}.config.error.${item.reason}`
-        ) || item.reason}`;
+        const lokaliseProm = this.hass
+          .loadBackendTranslation("config", item.domain)
+          .then(
+            (localize) =>
+              localize(
+                `component.${item.domain}.config.error.${item.reason}`
+              ) || item.reason
+          );
+        stateTextExtra = html`${until(lokaliseProm, item.reason)}`;
       } else {
         stateTextExtra = html`
           <br />
```

Comment (gjohansson-ST) [reply src/panels/config/integrations/ha-config-integration-page.ts]: I changed somehow to your suggestion but I haven't tested if it works yet

**Comment by gjohansson-ST** [comment]
Right now this is hidden behind the icon so it's not really obvious.
How about maybe change this so it would display the error below the initial error text instead of hiding in a mouse-over?

Current: 
Screenshot 2024-03-25 at 20 59 05

File: src/panels/config/integrations/ha-config-integration-page.ts [hunk 2]
```diff
@@ -550,10 +551,24 @@ class HaConfigIntegrationPage extends SubscribeMixin(LitElement) {
...
       if (item.reason) {
-        this.hass.loadBackendTranslation("config", item.domain);
-        stateTextExtra = html`${this.hass.localize(
-          `component.${item.domain}.config.error.${item.reason}`
-        ) || item.reason}`;
+        const lokalisePromError = this.hass
+          .loadBackendTranslation("config", item.domain)
+          .then((localize) =>
+            localize(`component.${item.domain}.config.error.${item.reason}`)
+          );
```
Comment (bramkragten) [inline src/panels/config/integrations/ha-config-integration-page.ts:558]: Why do we load this when we are only gonna use it if there is no `error_reason_translation_key`? Why not load it in the else?

**Review by bramkragten — APPROVED** [review]

File: src/panels/config/integrations/ha-config-integration-page.ts [hunk 3]
```diff
@@ -550,10 +551,24 @@ class HaConfigIntegrationPage extends SubscribeMixin(LitElement) {
         `ui.panel.config.integrations.config_entry.state.${item.state}`,
       ];
       if (item.reason) {
-        this.hass.loadBackendTranslation("config", item.domain);
-        stateTextExtra = html`${this.hass.localize(
-          `component.${item.domain}.config.error.${item.reason}`
-        ) || item.reason}`;
```
Comment (bdraco) [inline src/panels/config/integrations/ha-config-integration-page.ts:556]: The fallback to `|| item.reason` was removed in this PR

Comment (silamon) [reply src/panels/config/integrations/ha-config-integration-page.ts:556]: It probably resolves to an empty string?

Comment (bdraco) [reply src/panels/config/integrations/ha-config-integration-page.ts:556]:
```
{
    "id": 54,
    "type": "event",
    "event": [
        {
            "type": "updated",
            "entry": {
                "entry_id": "727e03e67745bd15d9923feb667cc0bf",
                "domain": "openweathermap",
                "title": "OpenWeatherMap",
                "source": "user",
                "state": "setup_retry",
                "supports_options": true,
                "supports_remove_device": false,
                "supports_unload": true,
                "supports_reconfigure": false,
                "pref_disable_new_entities": false,
                "pref_disable_polling": false,
                "disabled_by": null,
                "reason": "Please note that using One Call 3.0 requires a separate subscription to the One Call by Call plan. Learn more here https://openweathermap.org/price. If you have a valid subscription to the One Call by Call plan, but still receive this error, then please see https://openweathermap.org/faq#error401 for more info.",
                "error_reason_translation_key": null,
                "error_reason_translation_placeholders": null
            }
        }
    ]
}
```

reason is definitely populated coming back from the websocket api

Comment (bdraco) [reply src/panels/config/integrations/ha-config-integration-page.ts:556]: `item.reason` is populated

Comment (bdraco) [reply src/panels/config/integrations/ha-config-integration-page.ts:556]: `localize(`component.${item.domain}.config.error.${item.reason}`) || item.reason` fixes the issue

Comment (bdraco) [reply src/panels/config/integrations/ha-config-integration-page.ts:556]: https://github.com/home-assistant/frontend/pull/20668

**Pull request merged**